docs/solid-testing-library/api.mdx

```diff
@@ -0,0 +1,180 @@
+---
+id: api
+title: API
+sidebar_label: API
+---
+
+Due to being inspired by the [preact-testing-library][preact-docs] you can check
+its page for more information.
+
+[preact-docs]: ../preact-testing-library/intro.mdx
+
+There are several key differences, to be aware of.
+
+## Differences from the Preact Testing Library
+
+### `render`
+
+The `render` function takes in a function that returns a Solid Component, rather
+than simply the component itself.
+
+```tsx
+// With @testing-library/preact
+const results = render(<YourComponent />, options)
+```
+
+```tsx
+// With solid-testing-library
+const results = render(() => <YourComponent />, options)
+```
+
+Solid.js does _not_ re-render, it merely executes side effects triggered by
+reactive state that change the DOM, therefore there is no `rerender` method. You
+can use global signals to manipulate your test component in a way that causes it
+to update.
+
+In extension of the original API, the render function of this testing library
+supports a convenient `location` option that will set up an in-memory router
+pointing at the specified location. Since this setup is not instantaneous, you
+need to first use asynchronous queries (`findBy`) after employing it:
+
+```tsx
+it('uses params', async () => {
+  const App = () => (
+    <Routes>
+      <Route path="/ids/:id" component={() => <p>Id: {useParams()?.id}</p>} />
+      <Route path="/" component={() => <p>Start</p>} />
+    </Routes>
+  )
+  const {findByText} = render(() => <App />, {location: 'ids/1234'})
+  expect(await findByText('Id: 1234')).not.toBeFalsy()
+})
+```
+
+It uses `@solidjs/router`, so if you want to use a different router, you should
+consider the `wrapper` option instead. If you attempt to use this without having
+the package installed, you will receive an error message.
+
+### `renderHook`
+
+Solid.js external reactive state does not require any DOM elements to run in, so
+our `renderHook` call to test hooks in the context of a component (if your hook
+does not require the context of a component, `createRoot` should suffice to test
+the reactive behavior; for convenience, we also have `createEffect`, which is
```

Comment:
Maybe we should link to `createEffect` here? and then we'll be able to remove the "which is described later" sentence.

Reply:
will do :)